action_plugins/load_profile/__init__.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8; -*-
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from __future__ import annotations
+
+import enum
+import logging
+import threading
+import time
+from typing import List, TYPE_CHECKING
+from xml.etree import ElementTree
+
+from PySide6 import QtCore
+from PySide6.QtCore import Property, Signal, Slot
+
+from gremlin import error, event_handler, input_devices, \
+    joystick_handling, mode_manager, shared_state, util
+from gremlin.base_classes import AbstractActionData, AbstractFunctor, \
+    DataCreationMode, Value
+from gremlin.config import Configuration
+from gremlin.profile import Library
+from gremlin.types import ActionProperty, AxisMode, InputType, PropertyType
+
+from gremlin.ui.action_model import SequenceIndex, ActionModel
+
+if TYPE_CHECKING:
+    from gremlin.ui.profile import InputItemBindingModel
+
+
+class LoadProfileFunctor(AbstractFunctor):
+
+    """Executes a load profile action callback."""
+
+    def __init__(self, action: LoadProfileData):
+        super().__init__(action)
+
+        self._profile_filename = action.profile_filename
+
+    def __call__(
+        self,
+        event: event_handler.Event,
+        value: Value
+    ) -> None:
+
+        if not self._should_execute(value):
+                return
```

Comment:
That indentation seems too deep

Reply:
Will fix the indent and other unused imports as a part of cleanup.